### PR TITLE
define cosine similarity function

### PR DESCRIPTION
**Pull request by jakubhejhal**
Fixes issue https://github.com/autodistill/autodistill/issues/137

### CODE REVIEW

File: autodistill/core/embedding_ontology.py [hunk 1]
```diff
@@ -13,6 +13,19 @@
...
+    Calculate the cosine similarity between two vectors.
+
+    Args:
+        a: The first vector.
+        b: The second vector.
+
+    Returns:
+        The cosine similarity between the two vectors.
+    """
+    return float(np.dot(a, b) / (np.linalg.norm(a) * np.linalg.norm(b)))
```
Comment (lakshmanmoiiai) [inline autodistill/core/embedding_ontology.py:27]: Is it possible to do all the computations in torch 

    a = torch.from_numpy(a)
    b = torch.from_numpy(b)

    return float(torch.dot(a, b) / (torch.norm(a) * torch.norm(b))) somthing like as you don't have to move the embeddings to np and also GPU computation might be efficient

Comment (lakshmanmoiiai) [reply autodistill/core/embedding_ontology.py:27]: can you sign the cla?

Comment (lakshmanmoiiai) [reply autodistill/core/embedding_ontology.py:27]: just one more point, if the generated vectors return a zero norm in edge cases the cosine similarity will return Inf can you add
epsilon = 1e-10
cosine_similarity = float(np.dot(a, b) / (np.linalg.norm(a) * np.linalg.norm(b) + epsilon))

Comment (lakshmanmoiiai) [reply autodistill/core/embedding_ontology.py]: Looks good now

**Review by lakshmanmoiiai — APPROVED** [review]

**Comment by capjamesg** [comment]
@lakshmanmoiiai Thank you for your contribution! Your code looks good, although we need you to accept the CLA before we can merge it. Please accept the CLA at: https://github.com/autodistill/autodistill/pull/138#issuecomment-1961241363

**Comment by capjamesg** [comment]
Following up on this :)

**Comment by capjamesg** [comment]
A fix for this was introduced in #163, so I am going to close this PR. If you want to replace the current solution, feel free to submit a new PR.

**Pull request closed without merging**